doc/tools/docgram/orderedGrammar

```diff
@@ -2117,29 +2117,32 @@ rewrite_occs: [
 ]
 
 rewstrategy: [
-| one_term
 | "<-" one_term
-| "fail"
-| "id"
-| "refl"
-| "progress" rewstrategy
-| "try" rewstrategy
-| rewstrategy ";" rewstrategy
-| "choice" LIST1 rewstrategy
-| "repeat" rewstrategy
-| "any" rewstrategy
-| "subterm" rewstrategy
 | "subterms" rewstrategy
+| "subterm" rewstrategy
 | "innermost" rewstrategy
 | "outermost" rewstrategy
 | "bottomup" rewstrategy
 | "topdown" rewstrategy
+| "progress" rewstrategy
+| "try" rewstrategy
+| "any" rewstrategy
+| "repeat" rewstrategy
+| "old_hints" ident
 | "hints" ident
 | "terms" LIST0 one_term
 | "eval" red_expr
 | "fold" one_term
-| "(" rewstrategy ")"
-| "old_hints" ident
+| "choice" LIST1 rewstrategy_atom
+| rewstrategy_atom
+]
+
+rewstrategy_atom: [
+| one_term
+| "id"
+| "fail"
+| "refl"
```

Comment:
Put fail before id, SVP to match the earlier order (though it will still show up as diffs).

Reply:
OK, I understand my confusion. I tried to make the order in `orderedGrammar` preserve the order in `g_rewrite.mlg` but there were originally different. So, a decision has to be taken about what it means to match the earlier order: does it mean keeping the original order file per file or not changing the order for a given file (but uniformizing the order between files). I adopted the former (which I presume is what meant).

Moreover, the automatically-built order in `orderedGrammar`/`generalized-rewriting.rst` does not respect the one of `fullGrammar`. That is, I modified manually what was automatically generated for `orderedGrammar`.

Reply:
It looks good now.  You should generally ignore the order in fullGrammar and preserve the order in orderedGrammar, which the text should match.  Unless you have a good reason to change the order in the doc.

Sometimes fullGrammar productions have to be in a specific order to parse correctly in all cases, which may not be the best way to present the information to users.